### DISCUSSION TITLE
FontBakery Summit 2024

### DISCUSSION THREAD

**Discussion by davelab6** [fonttools/fontbakery#4169] [General]
I would like @felipesanches to organize a "fontbakery summit" event with all current and prospective organizations that use and contribute this project.

Would this summer be a good moment to make it happen? 

Perhaps timed around TypeCon in Portland Oregon in August, so there can be a face-to-face aspect?

**Comment by felipesanches** [comment]
<strike>I would suggest having an online meeting (possibly on Monday, August 7th) before organizing an in-person one.</strike>

**Please see: https://github.com/googlefonts/fontbakery/discussions/4169#discussioncomment-6627557**

**Comment by guidoferreyra** [comment]
+1

**Comment by graphicore** [comment]
I'm on vacation then (Aug. 4th to Aug. 20th) but I'll see if I can join from there.

**Comment by eliheuer** [comment]
Doing something around TypeCon would be nice if possible, it could be really laid back and informal, just a FontBakery Meetup.

**Reply by felipesanches** [reply]
yes, I'll be at TypeCon. Let's get together there. The originally intended date of August 7th would now be too close and I was not able to properly organize this yet. So, an initial informal conversation at TypeCon might help me organize something better for later.

The most important topic I'd like to bring up in this gathering would be the need to improve the processes of determination of which checks are included on the Universal profile, which is the profile shared by everyone.

I think it is desirable to keep vendor-specific profiles growing quickly, while Universal requires more caution to evaluate what is really agreed upon by the majority of stakeholders. So, while the tool is, in practice, useful for the needs of each individual vendor, the longer term mission of the project is to put the extra effort to strengthen the universal profile.

But I feel we could still improve in the way we do it and that perhaps these processes could build upon the experience of standards organizations like the W3C in the way they manage proposals from a diverse set of players in the industry.

I think we should, for instance, appoint "owners" for each vendor-specific profile. And then have some system of voting (or vouching for) whenever a new check is proposed. So that these owners formally tell the community that they agree (or disagree) with inclusion of said check in the Universal profile.

Also, I feel that the milestones focused on point-releases that we have right now  have not been so much useful. Perhaps we should move to milestones (or maybe issue tags) based on a funneling pipeline, in which a check proposal (or a bug report for an existing check) goes through several stages of gradual progress in understanding of the problem, until they're ready for actual implementation.

Such stages might look like:
* 1 - Someone just sent an initial proposal and we're starting to discuss it, to try to grasp all details.
* 2 - A profile owner (or anyone volunteering to implement it) acknowledges that the problem was properly understood
* 3 - The check proposal includes a clear rationale text (which is probably a compilation of any conversation that happened on previous steps)
* 4 - Font samples are made available and used on the implementation of code-tests
* 5 - And implementation if available on an open pull request but tests are still not passing.
* 6 - The PR has a green build and is waiting for a code-review

The steps I listed above are just a draft of what I consider to be the typical (or desired) evolution of a check proposal. I'm not sure these are exactly the best way to slice the funnel into stages, though. Feedback is appreciated.

**Reply by davelab6** [reply]
@felipesanches how did it go at typecon?

**Reply by felipesanches** [reply]
It was super good! We had conversations with many people who use FontBakery and some other who do not use it yet, but definitely heard about it and are planning to give it a try.

We had in-person code-reviews and @eliheuer finished his PR that implements a new check.

We also now got FontBakery hosted on the FontTools github organization, which should facilitate conveying to users the idea that FontBakery is a tool for everyone, not just people working focused on onboarding on Google Fonts.

**Comment by davelab6** [comment]
I'd like to propose another summit online in Q1 :)
